source/databricks/calculation_engine/tests/calculation_logic/features/energy_calculations/positive_grid_loss/when_period_is_one_hour/states/scenario_state.py

```diff
@@ -0,0 +1,125 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from datetime import datetime
+
+from pyspark.sql import DataFrame, SparkSession
+
+import calculation_logic.utils as cl
+import calculation_logic.utils.factories as clf
+from package.codelists import (
+    QuantityQuality,
+    InputMeteringPointType,
+    InputSettlementMethod,
+    MeteringPointResolution,
+)
+
+
+def get_expected(*args) -> DataFrame:
+    """
+    This function can be used to custom build the expected results (dataframe).
+    It is also used a reference to locate the test scenario.
+    """
+    return cl.create_energy_result_dataframe(*args)
+
+
+def create_grid_loss_metering_points(spark: SparkSession) -> DataFrame:
+    factory = clf.InputGridLossTestFactory(spark)
+    row1 = factory.create_row(metering_point_id="571313180400100657")
+    row2 = factory.create_row(metering_point_id="571313180480500149")
+    return factory.create_dataframe([row1, row2])
+
+
+def create_time_series_points(spark: SparkSession) -> DataFrame:
+    factory = clf.InputTimeSeriesPointTestFactory(spark)
+    row1 = factory.create_row(
+        metering_point_id="571313180400010437",
+        quantity=10.0,
+        quality=QuantityQuality.MEASURED.value,
+        observation_time=datetime(2023, 2, 1, 12, 0, 0),
+    )
+    row2 = factory.create_row(
+        metering_point_id="571313180400010673",
+        quantity=5.0,
+        quality=QuantityQuality.MEASURED.value,
+        observation_time=datetime(2023, 2, 1, 12, 0, 0),
+    )
+    row3 = factory.create_row(
+        metering_point_id="571313180400140417",
+        quantity=10.0,
+        quality=QuantityQuality.MEASURED.value,
+        observation_time=datetime(2023, 2, 1, 12, 0, 0),
+    )
+    return factory.create_dataframe([row1, row2, row3])
+
+
+def create_metering_point_periods(spark: SparkSession) -> DataFrame:
+    factory = clf.InputMeteringPointPeriodsTestFactory(spark)
+    row1 = factory.create_row(
+        metering_point_id="571313180400100657",
+        metering_point_type=InputMeteringPointType.CONSUMPTION,
+        settlement_method=InputSettlementMethod.FLEX,
+        grid_area="804",
+        resolution=MeteringPointResolution.QUARTER,
+        energy_supplier="8100000000115",
+        balance_responsible_id="5790001270940",
+        from_date=datetime(2023, 1, 31, 23, 0, 0),
+    )
+
+    row2 = factory.create_row(
+        metering_point_id="571313180480500149",
+        metering_point_type=InputMeteringPointType.PRODUCTION,
+        grid_area="804",
+        resolution=MeteringPointResolution.QUARTER,
+        energy_supplier="8100000000108",
+        balance_responsible_id="8100000000207",
+        from_date=datetime(2023, 1, 31, 23, 0, 0),
+    )
+
+    row3 = factory.create_row(
```

Comment:
How about
```suggestion
    row3 = factory.create_exchange_row(
```
and then remove the from_grid_area and to_grid_area from the regular create_row function?

Reply:
New task.